src/docs/sdk/performance/opentelemetry/span-protocol.mdx

```diff
@@ -0,0 +1,342 @@
+}
+
+// canonicalCodesGrpcMap maps some GRPC codes to Sentry's span statuses. See description in grpc documentation.
+var canonicalCodesGrpcMap = map[string]sentry.SpanStatus{
+	"1":  sentry.SpanStatusCanceled,
+	"2":  sentry.SpanStatusUnknown,
+	"3":  sentry.SpanStatusInvalidArgument,
+	"4":  sentry.SpanStatusDeadlineExceeded,
+	"5":  sentry.SpanStatusNotFound,
+	"6":  sentry.SpanStatusAlreadyExists,
+	"7":  sentry.SpanStatusPermissionDenied,
+	"8":  sentry.SpanStatusResourceExhausted,
+	"9":  sentry.SpanStatusFailedPrecondition,
+	"10": sentry.SpanStatusAborted,
+	"11": sentry.SpanStatusOutOfRange,
+	"12": sentry.SpanStatusUnimplemented,
+	"13": sentry.SpanStatusInternalError,
+	"14": sentry.SpanStatusUnavailable,
+	"15": sentry.SpanStatusDataLoss,
+	"16": sentry.SpanStatusUnauthenticated,
+}
+
+code := spanStatus.Code()
+if code < 0 || int(code) > 2 {
+    return sentry.SpanStatusUnknown, fmt.Sprintf("error code %d", code)
+}
+httpCode, foundHTTPCode := tags["http.status_code"]
+grpcCode, foundGrpcCode := tags["rpc.grpc.status_code"]
+var sentryStatus sentry.SpanStatus
+switch {
+case code == 1 || code == 0:
+    sentryStatus = sentry.SpanStatusOK
+case foundHTTPCode:
+    httpStatus, foundHTTPStatus := canonicalCodesHTTPMap[httpCode]
+    switch {
+    case foundHTTPStatus:
+        sentryStatus = httpStatus
+    default:
+        sentryStatus = sentry.SpanStatusUnknown
+    }
+case foundGrpcCode:
+    grpcStatus, foundGrpcStatus := canonicalCodesGrpcMap[grpcCode]
+    switch {
+    case foundGrpcStatus:
+        sentryStatus = grpcStatus
+    default:
+        sentryStatus = sentry.SpanStatusUnknown
+    }
+default:
+    sentryStatus = sentry.SpanStatusUnknown
+}
+return sentryStatus
+```
+
+### Span Events
+
+OpenTelemetry, has the concept of [Span Events](https://github.com/open-telemetry/opentelemetry-proto/blob/724e427879e3d2bae2edc0218fff06e37b9eb46e/opentelemetry/proto/trace/v1/trace.proto#L193-L214). As per the spec:
+
+> An event is a human-readable message on a span that represents “something happening” during it’s lifetime
+
+In Sentry, we have two options for how to treat span events. First, we can add them as breadcrumbs to the transaction the span belongs to. Second, we can create an artificial "point-in-time" span (a span with 0 duration), and add it to the span tree. TODO on what approach we take here.
+
+In the special case that the span event is an exception span, [where the `name` of the span event is `exception`](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/exceptions/), we also have the possibility of generating a Sentry error from an exception. In this case, we can create this [exception based on the attributes of an event](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/exceptions/#attributes), which include the error message and stacktrace. This exception can also inherit all other attributes of the span event + span as tags on the event.
```

Comment:
which and how many attributes do we expect there? if there are a ton of default attributes attached to an exception span, can we define which of them actually end up in event tags?

there is a storage concern about adding too many default tags to the event payload, and a product one as well considering user-defined tags and tags inherited from otel count against the same size limit